### DISCUSSION TITLE
Home assistant add-on

### DISCUSSION THREAD

**Discussion by matthew73210** [tphakala/birdnet-go#55] [General]
Hey,

I've seen that you've started on a docker version of this. In the past i've managed to port a couple projects to the home assistant standard, all it needs is to use one of their S6-overlay images. And call the cmd through the S6 overlay instead of in the dockerfile. 
The overlay being based on debian its self if needed.
Are there any specifics that i should know of? The idea being that you can use ingress and access the webui through HA using a rtsp source.
I'll give it a go when i have some free time and a good internet connection. Currently struggling on dialup speeds in the middle of nowhere.

Cheers,

**Comment by isZumpo** [comment]
Originally when I wrote the dockerfile for this project, I wanted to make use of the same image setup as images from linuxserver. As I find those images really easy to use, and like the uid:guid setup through s6-overlay.

I never got around to doing it, but it seems like it should be possible to extend one of their base images: https://github.com/linuxserver/docker-baseimage-debian

Do you know if there is any advantage to using a base image from homeassistant compared to  linuxserver?

**Reply by matthew73210** [reply]
TBH i don't. But i can guess is that it's preconfigured to talk to the HA supervisor and just be ready to run.

On the S6 side of things you'd only need some thing like the following :


# Copy root filesystem
COPY rootfs /

# Set the entrypoint to start the S6 supervisor
ENTRYPOINT ["/init"]


And have a longshot run "realtime".
I think the only difference is the base image to build and some specifs that HA needs but don't affect how the app runs (user inputs, ports etc..)

**Reply by isZumpo** [reply]
I suppose that it does not hurt to try it out 😊Though I would be a bit careful with introducing any type of dependency towards homeassistant since not everyone is using it. Maybe we should produce two docker images, one with and one without homeassistant dependencies? Do you know if there is any other project with this ha integration that could be looked at for inspiration?

One last question, is this the base image you are refering to: https://github.com/home-assistant/docker-base

**Reply by matthew73210** [reply]
Hey, sorry i thourght that i had replied. Yes that seems to be the source. I finish work and will be moving to a new site with better internet (having trouble on dialup esque speeds) so i'll start trail and error then. As you mentioned it's probs easier to have to containers.
Birdcage comes to mind, thats where i helped they guy create a HA addon, but it used docker compose so was more of a hack.

**Reply by alexbelgium** [reply]
An addon would be great indeed and make usage of all the resources in the new rpi that are oversized for simply running HA :) I've done a few addons also in my repo so perhaps I can help if you have started something and are encountering some complexities! This project looks really great

**Reply by alexbelgium** [reply]
Btw using an HomeAssistant base image only adds some custom helper modules (bashio ; tempio) and startup scripts depending on those. It is much easier to start from a linuxserver base image to create a universal container ; then reference this universal container in the HA addon dockerfile and just add those elements.

**Reply by tphakala** [reply]
While I do use HomeAssistant I use it very minimally and have no experience in developing such add-on, would anyone here be willing to start working on this feature?

**Reply by matthew73210** [reply]
Yep I'll give it a go on my computer tomorrow, really you can just grab the tarballs and run the same command as you do. It should change anything. If I can get that going I'll post here. The good bit about the s6 is that you can run things as a service, so instead of restarting the container etc..

**Comment by tomlut** [comment]
While this would be a nice addition to Home Assistant my server is not in a location that microphone access would be possible.

How about publishing detections to an mqtt server?

**Reply by tphakala** [reply]
Please submit an feature request as issue and I will work on it at some point.

**Reply by tphakala** [reply]
Just to point out here that MQTT support was added in v0.5.1 release

**Comment by tomlut** [comment]
In the issues or discussion section?

**Reply by tphakala** [reply]
Issues, add enhancement label so that it is easily identifiable as feature request

**Comment by jasonmadigan** [comment]
For users of `lovelylain/hass_ingress`, I added a sample Birdnet-go ingress config. It mostly works (daily summary doesn't seem to load, so I'm missing something):

https://github.com/lovelylain/hass_ingress/discussions/31
